### PR TITLE
Adding incoming call events to swagger

### PR DESCRIPTION
**Pull request by osaghaso**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

File: specification/communication/data-plane/Common/stable/2022-07-13/common.json [hunk 1]
```diff
@@ -84,6 +84,30 @@
...
+      ],
+      "properties": {
+        "unknownIdentifiers": {
+          "$ref": "#/definitions/UnknownIdentifierModel",
+          "description": "The unknown identifier."
+        }
+      }
+    },
+    "UnknownIdentifierModel": {
+      "description": "Unknown identifier identifier model",
```
Comment (pgrandhi) [inline specification/communication/data-plane/Common/stable/2022-07-13/common.json:103]: typo: Unknown Communication Identifier model

File: package.json [hunk 2]
```diff
@@ -37,6 +37,10 @@
     "prettier-check": "ts-node ./scripts/prettier-check.ts",
     "prettier-fix": "prettier --write specification/**/*.json",
     "prettier": "prettier",
-    "tsc": "tsc"
+    "tsc": "tsc",
```
Comment (DominikMe) [inline package.json:40]: Can we revert the changes in package.json and package-lock.json? These apply to all Azure services

File: specification/communication/data-plane/Common/stable/2022-07-13/common.json [hunk 3]
```diff
@@ -84,6 +84,49 @@
         "modelAsString": true
       }
     },
+    "CallingCommunicationIdentifierModel": {
```
Comment (DominikMe) [inline specification/communication/data-plane/Common/stable/2022-07-13/common.json:87]: Why are we adding these models to common.json? Common.json is only used for shared models across ACS REST APIs.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
+{
+  "id": "e80026e7-e298-46ba-bc42-dab0eda92581",
+  "sorce": "/subscriptions/{subscription-id}/resourceGroups/{group-name}/providers/Microsoft.Communication/communicationServices/{communication-services-resource-name}",
```
Comment (DominikMe) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json:3]: Should this be `topic`?
```suggestion
  "topic": "/subscriptions/{subscription-id}/resourceGroups/{group-name}/providers/Microsoft.Communication/communicationServices/{communication-services-resource-name}",
```

**Review by DominikMe — CHANGES_REQUESTED** [review]

File: specification/communication/data-plane/Common/stable/2022-07-13/common.json (hunk 3 above)
Comment (osaghaso) [reply specification/communication/data-plane/Common/stable/2022-07-13/common.json:87]: @DominikMe I added it to the Common.js cause these models can be used again for a different acs events in the future.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json (hunk 4 above)
Comment (osaghaso) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json]: @DominikMe Following the syntax from the other example in the cloud-events-schema folder their topic was renamed to source.

File: specification/communication/data-plane/Common/stable/2022-07-13/common.json (hunk 3 above)
Comment (DominikMe) [reply specification/communication/data-plane/Common/stable/2022-07-13/common.json:87]: Let's not add them here. You would also need to create a new api-version for common and can't just modify.

File: specification/communication/data-plane/Common/stable/2022-07-13/common.json [hunk 5]
```diff
@@ -84,6 +84,49 @@
...
+        }
+      ],
+      "properties": {
+        "unknownIdentifiers": {
+          "$ref": "#/definitions/UnknownIdentifierModel",
+          "description": "The unknown identifier."
+        }
+      }
+    },
+    "UnknownIdentifierModel": {
```
Comment (DominikMe) [inline specification/communication/data-plane/Common/stable/2022-07-13/common.json:102]: There is already an unknown identifier in ACS common, it's an identifier that has type unknown and only populates the rawId. Can we delete this?

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json [hunk 6]
```diff
@@ -7,6 +7,40 @@
   },
   "paths": {},
   "definitions": {
+    "AcsIncomingCallEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for an Microsoft.Communication.IncomingCall event",
+      "type": "object",
+      "properties": {
+        "to": {
+          "description": "The communication identifier of the target user.",
+          "$ref": "../../../../../communication/data-plane/Common/stable/2022-07-13/common.json#/definitions/CallingCommunicationIdentifierModel"
```
Comment (DominikMe) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:16]: You should just use the `CommunicationIdentifierModel` from common and not create your own subtype.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json [hunk 7]
```diff
@@ -7,6 +7,40 @@
...
+        },
+        "from": {
+          "description": "The communication identifier of the user who initiated the call.",
+          "$ref": "../../../../../communication/data-plane/Common/stable/2022-07-13/common.json#/definitions/CallingCommunicationIdentifierModel"
```
Comment (DominikMe) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:20]: same here

File: specification/communication/data-plane/Common/stable/2022-07-13/common.json [hunk 8]
```diff
@@ -84,6 +84,49 @@
...
+      "description": "Unknown Communication Identifier model",
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The unknown id."
+        }
+      }
+    },
+    "CustomContextModel": {
```
Comment (DominikMe) [inline specification/communication/data-plane/Common/stable/2022-07-13/common.json:112]: Let's move this into the incoming_call.jsons

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json (hunk 4 above)
Comment (DominikMe) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json]: Right, cloud event uses source, and you've fixed the typo 👍🏻

File: specification/communication/data-plane/Common/stable/2022-07-13/common.json (hunk 5 above)
Comment (osaghaso) [reply specification/communication/data-plane/Common/stable/2022-07-13/common.json:102]: I removed it

File: specification/communication/data-plane/Common/stable/2022-07-13/common.json (hunk 8 above)
Comment (osaghaso) [reply specification/communication/data-plane/Common/stable/2022-07-13/common.json:112]: where is this located. the incoming call json?

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json [hunk 9]
```diff
@@ -819,6 +867,27 @@
...
+    "AcsCallingCommunicationIdentifierProperties": {
+      "description": "Schema for Calling communication identifier",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../communication/data-plane/Common/stable/2022-07-13/common.json#/definitions/CommunicationIdentifierModel"
+        }
+      ],
+      "properties": {
+        "unknownIdentifiers": {
```
Comment (DominikMe) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:880]: I don't get what this is for, the CommunicationIdentifierModel already allows you to express an unknown identifier like this:
```json
{
  "kind": "unknown",
  "rawId": "12345"
}
```

Can we just remove `AcsCallingCommunicationIdentifierProperties` model and use `CommunicationIdentifierModel` directly?

**Review by DominikMe — APPROVED** [review]

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json (hunk 9 above)
Comment (DominikMe) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json]: Yes, you need to change the implementation of what you send to conform with how identifiers work across ACS APIs. Thanks for updating!

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json [hunk 10]
```diff
@@ -7,6 +7,54 @@
...
+      "description": "Schema of the Data property of an EventGridEvent for an Microsoft.Communication.IncomingCall event",
+      "type": "object",
+      "properties": {
+        "to": {
+          "description": "The communication identifier of the target user.",
+          "$ref": "../../../../../communication/data-plane/Common/stable/2022-07-13/common.json#/definitions/CommunicationIdentifierModel"
+        },
+        "from": {
+          "description": "The communication identifier of the user who initiated the call.",
+          "$ref": "../../../../../communication/data-plane/Common/stable/2022-07-13/common.json#/definitions/CommunicationIdentifierModel"
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:20]: This is using the wrong version and causes duplicate schema errors. Please ensure that you are using the same version as other references in this file.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json [hunk 11]
```diff
@@ -7,6 +7,54 @@
...
+        "customContext": {
+          "description": "Custom Context of Incoming Call",
+          "type": "object",
+          "properties": {
+            "customHeaders": {
+              "type": "string",
+              "description": "Custom Headers for incoming call"
+            },
+            "sipHeaders": {
+              "type": "object",
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:39]: This is not valid - there is no definition for how the object is formed.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json [hunk 12]
```diff
@@ -7,6 +7,54 @@
...
+              "description": "Sip Headers for incoming call"
+            },
+            "voipHeaders": {
+              "type": "object",
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:43]: This is not valid - there is no definition for how the object is formed.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json [hunk 13]
```diff
@@ -7,6 +7,54 @@
...
+          "description": "The Id of the server call",
+          "type": "string"
+        },
+        "callerDisplayName": {
+          "description": "Display name of caller.",
+          "type": "string"
+        },
+        "customContext": {
+          "description": "Custom Context of Incoming Call",
+          "type": "object",
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:32]: Please add a definition for this type.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json [hunk 14]
```diff
@@ -7,6 +7,54 @@
...
+              "type": "object",
+              "description": "Sip Headers for incoming call"
+            },
+            "voipHeaders": {
+              "type": "object",
+              "description": "Voip Headers for incoming call"
+            }
+          }
+        },
+        "incomingCallContext": {
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:48]: Is this just a free form string?

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json [hunk 15]
```diff
@@ -7,6 +7,54 @@
...
+          "description": "Signed incoming call context.",
+          "type": "string"
+        },
+        "correlationId": {
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:52]: Is this a Guid? Is it input to any other client library APIs?

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json [hunk 16]
```diff
@@ -7,6 +7,54 @@
...
+      "properties": {
+        "to": {
+          "description": "The communication identifier of the target user.",
+          "$ref": "../../../../../communication/data-plane/Common/stable/2022-07-13/common.json#/definitions/CommunicationIdentifierModel"
+        },
+        "from": {
+          "description": "The communication identifier of the user who initiated the call.",
+          "$ref": "../../../../../communication/data-plane/Common/stable/2022-07-13/common.json#/definitions/CommunicationIdentifierModel"
+        },
+        "serverCallId": {
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:22]: Is this a Guid? Is it input to any other client library APIs?

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json [hunk 17]
```diff
@@ -7,6 +7,54 @@
   },
   "paths": {},
   "definitions": {
+    "AcsIncomingCallEventData": {
+      "description": "Schema of the Data property of an EventGridEvent for an Microsoft.Communication.IncomingCall event",
+      "type": "object",
+      "properties": {
+        "to": {
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:14]: The naming is not consistent with other properties that use the CommunicationIdentifierModel.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/event-grid-schema/incoming_call.json [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+      "kind": "communicationUser",
+      "rawId": "{caller-id}",
+      "communicationUser": {
+        "id": "{caller-id}"
+      }
+    },
+    "serverCallId": "{server-call-id}",
+    "callerDisplayName": "VOIP Caller",
+    "customContext": {
+      "sipHeaders": {
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/event-grid-schema/incoming_call.json:23]: What about customHeaders?

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json (hunk 17 above)
Comment (osaghaso) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:14]: i dont understand

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json (hunk 14 above)
Comment (osaghaso) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:48]: what do you mean free string

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json (hunk 13 above)
Comment (osaghaso) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:32]: I dont understand

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json (hunk 17 above)
Comment (JoshLove-msft) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:13]: Other properties using this type are suffixed with CommunicationIdentifier, e.g. https://github.com/Azure/azure-rest-api-specs/blob/a4ad2d15d8086c162d48ef1c8cdb9118d8128ed8/specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json#L350

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json (hunk 13 above)
Comment (JoshLove-msft) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:31]: This needs to be its own definition, e.g. https://github.com/Azure/azure-rest-api-specs/pull/23762/files#diff-4b49e91c6d3dc4a0bf20909734cce568db54e4721a79a675e92c1dcb6440c53eR858

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json (hunk 15 above)
Comment (osaghaso) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:57]: not a guid

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json (hunk 16 above)
Comment (osaghaso) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:21]: not a guid

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/event-grid-schema/incoming_call.json (hunk 18 above)
Comment (osaghaso) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/event-grid-schema/incoming_call.json:23]: Its is optional it can be available or not available.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json (hunk 14 above)
Comment (JoshLove-msft) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:53]: Meaning that it is unstructured data.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json (hunk 13 above)
Comment (osaghaso) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:31]: thats what i had initailly but was told to add it here. changing back

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json (hunk 17 above)
Comment (osaghaso) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:13]: they use CommunicationIdentifierModel 
image

Comment (JoshLove-msft) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:13]: Yes, the comment is about the name of the property. Should it be suffixed with CommunicationIdentifier? Also, see the other naming comments left in the Api View about these properties.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/event-grid-schema/incoming_call.json (hunk 18 above)
Comment (JoshLove-msft) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/event-grid-schema/incoming_call.json:23]: It would be helpful to include it, especially since there is a question about whether it should be a dictionary.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json (hunk 14 above)
Comment (osaghaso) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:33]: yes free string

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json (hunk 17 above)
Comment (osaghaso) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:13]: ooh I see. I believe the goal was to treat it for easy as possible to just be incoming call from and incoming call to. But i will follow up with the call team to get clarification

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/event-grid-schema/incoming_call.json [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+        "userToUser": "616d617a6f6e5f6368696;encoding=hex",
+        "X-MS-Custom-myhader1": "35567842",
+        "X-MS-Custom-myhader2": "customsipheadervalue"
```
[the remark this answers is not in the document]
Comment (pgrandhi) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/event-grid-schema/incoming_call.json:26]: X-MS-Custom-Myheader2

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/event-grid-schema/incoming_call.json [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+      "communicationUser": {
+        "id": "{caller-id}"
+      }
+    },
+    "serverCallId": "{server-call-id}",
+    "callerDisplayName": "VOIP Caller",
+    "customContext": {
+      "sipHeaders": {
+        "userToUser": "616d617a6f6e5f6368696;encoding=hex",
+        "X-MS-Custom-myhader1": "35567842",
```
[the remark this answers is not in the document]
Comment (pgrandhi) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/event-grid-schema/incoming_call.json:25]: X-MS-Custom-Myheader1

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json [hunk 21]
```diff
@@ -0,0 +1,38 @@
...
+      "communicationUser": {
+        "id": "{caller-id}"
+      }
+    },
+    "serverCallId": "{server-call-id}",
+    "callerDisplayName": "VOIP Caller",
+    "customContext": {
+      "sipHeaders": {
+        "userToUser": "616d617a6f6e5f6368696;encoding=hex",
+        "X-MS-Custom-myhader1": "35567842",
```
[the remark this answers is not in the document]
Comment (pgrandhi) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json:25]: X-MS-Custom-Myheader1

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json [hunk 22]
```diff
@@ -0,0 +1,38 @@
...
+        "X-MS-Custom-myhader2": "customsipheadervalue"
```
[the remark this answers is not in the document]
Comment (pgrandhi) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json:26]: X-MS-Custom-Myheader2

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json [hunk 23]
```diff
@@ -0,0 +1,26 @@
+{
+  "id": "81fa5460-7d8c-4fb7-bbb7-2d56154e78d9",
+  "topic": "/subscriptions/{subscription-id}/resourcegroups/} {group-name}/providers/microsoft.communication/communication services/{communication-services-resource-name}",
+  "subject": "/call/{call-id}",
```
[the remark this answers is not in the document]
Comment (osaghaso) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json]: addressed it

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json (hunk 21 above)
[the remark this answers is not in the document]
Comment (osaghaso) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json]: addressed it

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json (hunk 22 above)
[the remark this answers is not in the document]
Comment (osaghaso) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/examples/cloud-events-schema/incoming_call.json]: addressed it

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json [hunk 24]
```diff
@@ -819,6 +854,30 @@
           "type": "string"
         }
       }
+    },
+    "AcsIncomingCallCustomContext": {
+      "description": "Custom Context of Incoming Call",
+      "type": "object",
+      "properties": {
+        "customHeaders": {
+          "type": "string",
```
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:863]: Are we sure this is not meant to be a dictionary like sipHeaders and voipHeaders?

**Comment by JoshLove-msft** [comment]
Looks like there are merge conflicts, can you please merge from main or rebase?

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json (hunk 24 above)
Comment (osaghaso) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:863]: yes

**Review by JoshLove-msft — APPROVED** [review]
